3520/0/about_variable_scope.rb

```diff
@@ -0,0 +1,116 @@
+# rubocop:disable all
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+# This class smells of :reek:UncommunicativeMethodName and :reek:ClassVariable
+# This class smells of :reek:UncommunicativeParameterName and :reek:UncommunicativeVariableName
+class AboutVariableScope < Neo::Koan
+
+  def bark
+    noise = 'RUFF'
+  end
+
+  def test_noise_is_not_available_in_the_current_scope
+
+    assert_raise(___) do
+      noise
+    end
+
+  end
+
+  def test_we_can_get_noise_by_calling_method
+    assert_equal __, bark
+  end
+
+  inaccessible = 'Outside our universe'
+  def test_defs_cannot_access_variables_outside_scope
+    # defined? does not return true or false
+    assert_equal __, defined? inaccesible
+  end
+
+  # ------------------------------------------------------
+
+  def test_blocks_can_access_variables_outside_scope
+    test = 'Hi'
+    (1..2).each do
+      test = 'Hey'
+    end
+
+    assert_equal __, test
+  end
+
+  def test_block_variables_cannot_be_accessed_outside_scope
+    (1..2).each do
+      x = 0
+    end
+    assert_equal __, defined? x
+  end
+
+  # ------------------------------------------------------
+
+  class Mouse
+    @@total = 0
+    # Class variables are prefixed with two '@' characters.
+
+    def initialize(n)
+      @name = n
+      # Instance variables are prefixed with one '@' character.
+      @@total += 1
+    end
+
+    def name
+      @name
+    end
+
+    def Mouse.count
+      @@total
+    end
+  end
+
+  def test_instance_variable
+    oscar = Mouse.new('Oscar')
+    assert_equal __, oscar.name
```

Comment:
Complete this tasks too

Reply:
This task does'n include in path, but I'll find out how to complete it.